### PR TITLE
doc: update pkg dependency matrix and misc. fixes

### PR DESCRIPTION
**Pull request by dongahn**
This PR updates the package dependency matrix
of our main README file and also adds fixes
for other misc documentation issue.

Fix Issue #647, #686 and #598.

### CODE REVIEW

File: resource/utilities/README.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ On start-up, `resource-query` reads in a resource-graph generation recipe
 written in the GRUG format (see the GRUG section below) and populates
 a resource-graph data store, representing HPC resources and their
 relationships according to Flux Resource Model
-([RFC 4](https://github.com/flux-framework/rfc/blob/master/spec_4.adoc)).
+([RFC 4](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_14.html)).
```
Comment (cmoussa1) [inline resource/utilities/README.md:27]: If this is supposed to point to RFC 4, I believe the hyperlink should be in fact be: https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_4.html. The hyperlink you included also takes me to RFC 14.

Comment (dongahn) [reply resource/utilities/README.md:27]: Great catch! That should have been 4. Will update and force a push.

**Comment by dongahn** [comment]
OK. Fix the incorrect link and forced a push. Thanks @cmoussa1!

**Comment by dongahn** [comment]
Thanks @cmoussa1!

**Pull request merged**